nebu/cli/get.py

```diff
@@ -116,37 +113,63 @@ def get(ctx, env, col_id, col_version, output_dir, book_tree, get_resources):
 
     # Write tree
     tree = col_metadata['tree']
-    os.mkdir(str(output_dir))
 
-    num_pages = _count_leaves(tree) + 1  # Num. of xml files to fetch
+    num_pages = _count_nodes(tree)
     try:
-        label = 'Getting {}'.format(output_dir.relative_to(Path.cwd()))
+        label = f'Getting {output_dir.relative_to(Path.cwd())}'
     except ValueError:
-        # Raised ONLY when output_dir is not a child of cwd
-        label = 'Getting {}'.format(output_dir)
+        # Raised when output_dir is not a child of cwd
+        label = f'Getting {output_dir}'
     with click.progressbar(length=num_pages,
                            label=label,
                            width=0,
                            show_pos=True) as pbar:
-        _write_node(tree, base_url, output_dir, book_tree, get_resources, pbar)
+        loop = asyncio.get_event_loop()
+        loop.set_exception_handler(report_and_quit)
+        coro = _write_contents(tree,
+                               base_url,
+                               output_dir,
+                               book_tree,
+                               get_resources,
+                               pbar)
+        loop.run_until_complete(coro)
 
 
-def _count_leaves(node):
-    if 'contents' in node:
-        return sum([_count_leaves(child) for child in node['contents']])
-    else:
-        return 1
+def report_and_quit(loop, context):  # pragma: no cover
+    loop.default_exception_handler(context)
 
+    exception = context.get('exception')
+    print_tb(exception.__traceback__)
+    print(type(exception))
+    print(str(exception))
+    loop.stop()
```

Comment:
It will because the event loop will terminate before the futures complete, which causes an actual exception. I could just raise the exception explicitly though...

Reply:
Haha just kidding. That was an absolute nightmare. This is definitely the easiest way. I have confirmed a RuntimeError occurs which should cause non-zero status code. I'm not sure I can do better than that.

Reply:
Haha just kidding. That was an absolute nightmare. This is definitely the easiest way. I have confirmed a RuntimeError occurs which should cause non-zero status code. I'm not sure I can do better than that.